platform/mellanox/mlnx-platform-api/sonic_platform/chassis.py

```diff
@@ -95,5 +129,169 @@ def _get_port_position_tuple_by_sku_name(self):
+            'XX:XX:XX:XX:XX:XX'
+        """
+        return self.eeprom.get_base_mac()
+
+    def get_serial_number(self):
+        """
+        Retrieves the hardware serial number for the chassis
+
+        Returns:
+            A string containing the hardware serial number for this chassis.
+        """
+        return self.eeprom.get_serial_number()
+
+    def get_system_eeprom_info(self):
+        """
+        Retrieves the full content of system EEPROM information for the chassis
+
+        Returns:
+            A dictionary where keys are the type code defined in
+            OCP ONIE TlvInfo EEPROM format and values are their corresponding
+            values.
+        """
+        return self.eeprom.get_system_eeprom_info()
+
+    def _read_generic_file(self, filename, len):
+        """
+        Read a generic file, returns the contents of the file
+        """
+        r = ''
+        try:
+            f = io.open(filename)
+            r = f.read(len)
+            f.close()
+            return r
+        except:
+            return ''
+
+    def _verify_reboot_cause(self, filename):
+        '''
+        Open and read the reboot cause file in 
+        /var/run/hwmanagement/system (which is defined as REBOOT_CAUSE_ROOT)
+        If a reboot cause file doesn't exists, returns '0'.
+        '''
+        return bool(int(self._read_generic_file(join(REBOOT_CAUSE_ROOT, filename), REBOOT_CAUSE_FILE_LENGTH).rstrip('\n')))
+
+    def get_reboot_cause(self):
+        """
+        Retrieves the cause of the previous reboot
+
+        Returns:
+            A tuple (string, string) where the first element is a string
+            containing the cause of the previous reboot. This string must be
+            one of the predefined strings in this class. If the first string
+            is "REBOOT_CAUSE_HARDWARE_OTHER", the second string can be used
+            to pass a description of the reboot cause.
+        """
+        #read reboot causes files in the following order
+        minor_cause = ''
+        if self._verify_reboot_cause(REBOOT_CAUSE_POWER_LOSS_FILE):
+            major_cause = self.REBOOT_CAUSE_POWER_LOSS
+        elif self._verify_reboot_cause(REBOOT_CAUSE_THERMAL_OVERLOAD_ASIC_FILE):
+            major_cause = self.REBOOT_CAUSE_THERMAL_OVERLOAD_ASIC
+        elif self._verify_reboot_cause(REBOOT_CAUSE_WATCHDOG_FILE):
+            major_cause = self.REBOOT_CAUSE_WATCHDOG
+        else:
+            major_cause = self.REBOOT_CAUSE_HARDWARE_OTHER
+            if self._verify_reboot_cause(REBOOT_CAUSE_MLNX_FIRMWARE_RESET):
+                minor_cause = "firmware"
+            else:
+                reboot_by_user = self._read_generic_file(PREVIOUS_REBOOT_CAUSE_FILE, PREVIOUS_REBOOT_CAUSE_MAX_LENGTH)
+                if '' != reboot_by_user:
+                    major_cause = self.REBOOT_CAUSE_NON_HARDWARE
+                    minor_cause = reboot_by_user
+
+        return major_cause, minor_cause
```

Comment:
This function is expected to simply return a string, one of the predefined "REBOOT_CAUSE_*" strings, not a tuple. If there are additional reboot causes you would like to handle, we can add more "REBOOT_CAUSE_*" strings. We would like to unify the SONiC experience among all platforms. If you believe there are Mellanox-specific reboot causes that for some reason shouldn't be added to the base class, we can revisit this approach.

Reply:
Hi Joe.
I got your idea. But according to the base class definition in sonic-platform-common, get_reboot_cause is supposed to return a tuple. Could you confirm it? 
You can find it here [https://github.com/Azure/sonic-platform-common/blob/master/sonic_platform_base/chassis_base.py#L90](url)

Reply:
Hi Stephen. You are correct. I forgot about the design change! Please ignore this comment. :)